src/pds/ingress/pds-nucleus-fz-to-fits-converter.py

```diff
@@ -0,0 +1,136 @@
+"""
+=========================================
+pds-nucleus-fz-to-fits-converter.py
+=========================================
+
+Lambda function to extract fz files (to get .fits files).
+This lambda function is triggered by pds-nucleus-s3-event-lambda-invoker lambda function when a
+new .fz file is copied to the staging S3 bucket.
+
+"""
+
+import json
+import urllib.parse
+import logging
+import shutil
+import boto3
+import os
+import time
+from xml.dom import minidom
+
+
+s3 = boto3.client('s3')
+logger = logging.getLogger("pds-nucleus-product-writer-logger")
+
+# Read environment variables from lambda configurations
+efs_mount_path = os.environ.get('EFS_MOUNT_PATH')
+db_clust_arn = os.environ.get('DB_CLUSTER_ARN')
+db_secret_arn = os.environ.get('DB_SECRET_ARN')
+efs_mount_path = os.environ.get('EFS_MOUNT_PATH')
+
+rds_data = boto3.client('rds-data')
+
+# Main lambda handler
+
+
+def lambda_handler(event, context):
+
+    # Get the object from the event and show its content type
+    bucket = event['Records'][0]['s3']['bucket']['name']
+    key = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'], encoding='utf-8')
+
+    logger.setLevel(logging.DEBUG)
+    logger.addHandler(logging.StreamHandler())
+    logger.info(f"Lambda Request ID: {context.aws_request_id}")
+
+    copy_file_to_efs(bucket, key, efs_mount_path)
+
+
+# Copies a file from S3 to EFS
+def copy_file_to_efs(s3_bucket, s3_key, efs_mount_path):
+    path_of_product = s3_bucket + "/" + s3_key
+    download_dir = os.path.dirname('/tmp/' + path_of_product)
+    download_file_path = os.path.normpath('/tmp/' + path_of_product)
+
+    try:
+        # Download the file from S3 (only /tmp has permissions to download)
+        logger.info(f"File downloading to : {download_file_path}")
+        os.makedirs(download_dir, exist_ok=True)
+        s3.download_file(s3_bucket, s3_key, download_file_path)
+        logger.info(f"File downloaded: {download_file_path}")
+    except Exception as e:
+        logger.error(
+            f"Error downloading file from S3. s3_bucket: {s3_bucket}, s3_key: {s3_key}, efs_mount_path: {efs_mount_path}, Exception: {str(e)}")
+
+    # Move the file to the /mnt/data directory
+    destination_path = efs_mount_path + os.path.dirname(path_of_product)
+    destination_file = efs_mount_path + path_of_product
+
+    try:
+        os.makedirs(destination_path, exist_ok=True)
+
+        if os.path.isfile(os.path.normpath(destination_file)):
+            os.remove(os.path.normpath(destination_file))
+            logger.debug(f"Deleted existing file in: {os.path.normpath(destination_file)}")
+
+        shutil.move(download_file_path, os.path.normpath(destination_path))
+        logger.info(f"File moved to: {destination_path}")
+
+        # Unpack .fz files
+        if destination_file.lower().endswith(".fz"):
+
+            s3_key_renamed = s3_key.rstrip(",.fz")
+            logger.info(f'Renamed to {s3_key_renamed}')
```

Comment:
Not a big deal, but loggers should be called with `%s` interpolation. Using an f-string makes Python do the string substitution in all cases, even when the `info` category is turned off. With `%s`, if `info` is off, then the string interpolation is skipped. (Alex Dunn disagrees vehemently with this—so maybe it's a matter of taste.)

Reply:
@nutjob4life, I prefer the f-strings :-). I agree that using % will skip the string interpolation when logs are turned off.  However, in case of this specific lambda function, all the log messages are doing very little processing.

And it seems there are various opinions on using % vs f-strings in logs.

- [% vs f-strings for string formatting in logs](https://betterstack.com/community/guides/logging/python/python-logging-best-practices/#5-vs-f-strings-for-string-formatting-in-logs )
- [What is the best string formatting technique for logging in Python?](https://dev.to/izabelakowal/what-is-the-best-string-formatting-technique-for-logging-in-python-d1d)

Reply:
I'll be over here, knowing what the _right_ way is 😂